examples/layeredeffects/AlgWP.fst

```diff
@@ -82,8 +82,9 @@ let rec interp_as_wp #a (t : Alg.tree a [Read;Write]) : st_wp a =
 
 (* With handlers. Can only be done into []? See the use of `run`. *)
 let interp_as_wp2 #a #l (t : rwtree a l) : Alg (st_wp a) [] =
+  let t0 : Alg.tree a [Read; Write] = t in
   handle_with #a #(st_wp a) #[Read; Write] #[]
-              (fun () -> Alg?.reflect t)
+              (fun () -> Alg?.reflect t0)
```

Comment:
This is a proof regression, but I think the behavior is expected since this `handle_with` parameter is marked with `$`, and so, the argument to `reflect` must have the type expected by equality, and not subtyping. Earlier I think we were not checking it properly.